lib/spack/docs/getting_started.rst

```diff
@@ -37,1396 +37,177 @@ This will create a directory called ``spack``. Once you have cloned Spack, we re
-
-The compiler executables are listed under ``extra_attributes:compilers``, and are keyed by language.
-Once you save the file, the configured compilers will show up in the list displayed by ``spack compilers``.
-
-You can also add compiler flags to manually configured compilers. These flags should be specified in the
-``flags`` section of the compiler specification. The valid flags are ``cflags``, ``cxxflags``, ``fflags``,
-``cppflags``, ``ldflags``, and ``ldlibs``. For example:
-
-.. code-block:: yaml
-
-   packages:
-     gcc:
-       externals:
-       - spec: gcc@10.5.0 languages='c,c++,fortran'
-         prefix: /usr
-         extra_attributes:
-           compilers:
-             c: /usr/bin/gcc-10
-             cxx: /usr/bin/g++-10
-             fortran: /usr/bin/gfortran-10
-           flags:
-             cflags: -O3 -fPIC
-             cxxflags: -O3 -fPIC
-             cppflags: -O3 -fPIC
-
-These flags will be treated by Spack as if they were entered from
-the command line each time this compiler is used. The compiler wrappers
-then inject those flags into the compiler command. Compiler flags
-entered from the command line will be discussed in more detail in the
-following section.
-
-Some compilers also require additional environment configuration.
-Examples include Intel's oneAPI and AMD's AOCC compiler suites,
-which have custom scripts for loading environment variables and setting paths.
-These variables should be specified in the ``environment`` section of the compiler
-specification. The operations available to modify the environment are ``set``, ``unset``,
-``prepend_path``, ``append_path``, and ``remove_path``. For example:
-
-.. code-block:: yaml
-
-   packages:
-     intel-oneapi-compilers:
-       externals:
-       - spec: intel-oneapi-compilers@2025.1.0
-         prefix: /opt/intel/oneapi
-         extra_attributes:
-           compilers:
-             c: /opt/intel/oneapi/compiler/2025.1/bin/icx
-             cxx: /opt/intel/oneapi/compiler/2025.1/bin/icpx
-             fortran: /opt/intel/oneapi/compiler/2025.1/bin/ifx
-           environment:
-             set:
-               MKL_ROOT: "/path/to/mkl/root"
-             unset: # A list of environment variables to unset
-               - CC
-             prepend_path: # Similar for append|remove_path
-               LD_LIBRARY_PATH: /ld/paths/added/by/setvars/sh
-
-^^^^^^^^^^^^^^^^^^^^^^^
-Build Your Own Compiler
-^^^^^^^^^^^^^^^^^^^^^^^
-
-If you are particular about which compiler/version you use, you might wish to have Spack build it for you.
-For example:
+   $ /home/spack/.local/spack/opt/linux-icelake/tcl-8.6.12-6vo5hxeqw5plzd6gvzm74wlfz5stnzcv/bin/tclsh
+   >% echo "Hello world!"
+   Hello world!
 
-.. code-block:: console
-
-   $ spack install gcc@14+binutils
+This works because of how Spack builds packages, setting the ``RPATH`` of their dependencies, but using such a long absolute path is not the most ergonomic way to refer to an executable.
```

Comment:
```suggestion
This works because of how Spack builds packages, setting the ``RPATH`` of their dependencies, but using such a long absolute path is not the most convenient way to run an executable.
```